presto-native-execution/presto_cpp/main/types/tests/PrestoToVeloxConnectorTest.cpp

```diff
@@ -53,3 +72,107 @@ TEST_F(PrestoToVeloxConnectorTest, addDuplicates) {
           std::make_unique<HivePrestoToVeloxConnector>(kConnectorName)),
       fmt::format("Connector {} is already registered", kConnectorName));
 }
+
+TEST_F(PrestoToVeloxConnectorTest, icebergPreservesColumnNameCase) {
+  protocol::List<protocol::Column> dataColumns;
+  protocol::Column col1;
+  col1.name = "MixedCaseCol1";
+  col1.type = "integer";
+  dataColumns.push_back(col1);
+
+  protocol::Column col2;
+  col2.name = "UPPERCASECOL2";
+  col2.type = "varchar";
+  dataColumns.push_back(col2);
+
+  auto trueConstant = std::make_shared<protocol::ConstantExpression>();
+  trueConstant->type = "boolean";
+  trueConstant->valueBlock.data = "CgAAAEJZVEVfQVJSQVkBAAAAAAE=";  // base64-encoded true value.
+
+  auto icebergLayout =
+      std::make_shared<protocol::iceberg::IcebergTableLayoutHandle>();
+  icebergLayout->domainPredicate.domains = std::make_shared<
+      protocol::Map<protocol::Subfield, protocol::Domain>>();
+  icebergLayout->remainingPredicate = trueConstant;
+  icebergLayout->pushdownFilterEnabled = false;
+  icebergLayout->dataColumns = dataColumns;
+  icebergLayout->partitionColumns = {};
+  icebergLayout->predicateColumns = {};
+
+  protocol::TableHandle tableHandle;
+  tableHandle.connectorId = "iceberg";
+  auto icebergHandle =
+      std::make_shared<protocol::iceberg::IcebergTableHandle>();
+  icebergHandle->schemaName = "test_schema";
+  icebergHandle->icebergTableName.tableName = "test_table";
+  tableHandle.connectorHandle = icebergHandle;
+  tableHandle.connectorTableLayout = icebergLayout;
+
+  IcebergPrestoToVeloxConnector icebergConnector("iceberg");
+  connector::ColumnHandleMap assignments;
+  auto result = icebergConnector.toVeloxTableHandle(
+      tableHandle, *exprConverter_, *typeParser_, assignments);
+
+  ASSERT_NE(result, nullptr);
+  auto* handle = dynamic_cast<connector::hive::HiveTableHandle*>(result.get());
+  ASSERT_NE(handle, nullptr);
+
+  // Verify Iceberg preserves column name case.
+  auto dataColumnsType = handle->dataColumns();
+  ASSERT_NE(dataColumnsType, nullptr);
+  EXPECT_EQ(dataColumnsType->size(), 2);
+  EXPECT_EQ(dataColumnsType->nameOf(0), "MixedCaseCol1");
+  EXPECT_EQ(dataColumnsType->nameOf(1), "UPPERCASECOL2");
+}
```

Comment:
Both tests share a lot of common setup... Is it possible to combine and show that one preserves case while the other not.